components/nimbus/ios/Nimbus/JSONUtil.swift

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+internal func stringify(jsonObject: [String: Any]) throws -> String {
```

Comment:
nit: I don't know if you can create an extension for a dictionary type, but could this and `toJson` below be `internal extension`s for their respective types? That would make calling it instead of like this:
`let string = try stringify(jsonObject: additionalContext)`

To something like this:
`let string = try additionalContext.stringify()`

This seems to be a little more idomatic of Swift and doesn't change how this works, so feel free to disregard.